PAS_Database/dbo/Stored Procedures/USP_UpdateSOPartCostDetails.sql

```diff
@@ -166,7 +169,7 @@ SET NOCOUNT ON
 						SET UnitSalesPriceExtended = ISNULL(UnitSalesPrice, 0) * @PartQty,
 						UnitCostExtended = ISNULL(UnitCost, 0) * @PartQty,
 						NetSaleAmount = (ISNULL((ISNULL(UnitSalesPrice, 0) * @PartQty), 0) + MarkUpAmount) - DiscountAmount,
-						TotalRevenue = ((ISNULL((ISNULL(UnitSalesPrice, 0) * @PartQty), 0) + MarkUpAmount) - DiscountAmount) + MiscCharges
+						TotalRevenue = ((ISNULL((ISNULL(UnitSalesPrice, 0) * @PartQty), 0) + MarkUpAmount) - DiscountAmount) + ISNULL(@Charges_S, 0)
```

Comment:
Have you verified if we won't have any stockline and just charges added at part level? In that case @Charges_S will be NULL